### PR TITLE
Update NodeJS backend dependencies.

### PR DESCRIPTION
**Pull request by trev-dev**
Jira: HD-26946

The goal of this pull request is to update the NodeJS backend and its dependencies.  Due to some of the updates being incompatible with CommonJS modules, I opted to refactor the backend into the ESM (EcmaScript Module) format.  This is the standard for JavaScript, whereas CJS was more of a node/webpack/bundler thing.

Breaking changes for nats and passport have been addressed as well.  Check the commit messages, I tried to keep things granular.

Jest was also dropped in favor of Vitest.  Vitest is actually prepared to test ESM files.  This change removed a great deal of bloat from the dependency tree and sped up tests a great deal.

### CODE REVIEW

File: backend/src/components/utils.js [hunk 1]
```diff
@@ -29,23 +39,23 @@
...
 }
 
-function getAccessToken(req) {
+export function getAccessToken(req) {
   const user = getSessionUser(req);
   return user && user.jwt;
 }
 
-async function deleteData(token, url, correlationID) {
+export async function deleteData(token, url, correlationID) {
```
[the remark this answers is not in the document]
Comment (arcshiftsolutions) [reply backend/src/components/utils.js:58]: @trev-dev - Yeah we ran into the same thing - the CodeQL engine isn't always smart enough to see what we're doing in middleware.

File: backend/src/routes/auth.js [hunk 2]
```diff
@@ -94,6 +99,7 @@ router.get('/logout', async (req, res, next) => {
     } else {
       retUrl = makeUrl('/logout');
     }
+    console.log('RET ::: ', retUrl);
```
Comment (arcshiftsolutions) [inline backend/src/routes/auth.js:102]: ^^

File: backend/src/routes/auth.js [hunk 3]
```diff
@@ -65,12 +65,17 @@ addBaseRouterGet('oidcBceidUMP', '/login_bceid_ump');
...
-  const makeUrl = endpoint => encodeURIComponent(
-    config.get('logoutEndpoint')
-    + `?post_logout_redirect_uri=${config.get('server:frontend')}`
-    + endpoint
-    + `&client_id=${config.get('oidc:clientId')}`
-  );
+  let idToken = req?.session?.passport?.user?.idToken;
+
+  const makeUrl = endpoint => {
+    console.log(req.user);
```
Comment (arcshiftsolutions) [inline backend/src/routes/auth.js:71]: ^^

File: backend/src/routes/auth.js (hunk 2 above)
Comment (trev-dev) [reply backend/src/routes/auth.js:102]: Ugh, noooooo!  Sorry, nice catch.

**Review by arcshiftsolutions — APPROVED** [review]

**Pull request merged**